File: Algorithm-CPP/30_3의개수는.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int n, result = 0;
    cin >> n;

    int lt = -1, rt = -1;
    for(int i = 0; lt != 0; i++) {
        int current = (n % int(pow(10, i + 1))) / pow(10, i);
        lt = n / pow(10, i + 1);
        rt = n % int(pow(10, i));

        if(current > 3) result += (lt + 1) * pow(10, i);
        else if(current < 3) result += lt * pow(10, i);
        else result += lt * pow(10, i) + (rt + 1);
    }

    cout << result << endl;

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/30_3의개수는.swift
//
//  30_3의개수는.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/08.
//

import Foundation

func solution30() {
    let n = Int(readLine()!)!
    var lt = -1, rt = -1, i = 0, result = 0

    while lt != 0 {
        let current = (n % Int(pow(Double(10), Double(i + 1)))) / Int(pow(Double(10), Double(i)))
        lt = n / Int(pow(Double(10), Double(i + 1)))
        rt = n % Int(pow(Double(10), Double(i)))

        if current > 3 {
            result += (lt + 1) * Int(pow(Double(10), Double(i)))
        } else if current < 3 {
            result += lt * Int(pow(Double(10), Double(i)))
        } else {
            result += lt * Int(pow(Double(10), Double(i))) + (rt + 1)
        }

        i += 1
    }

    print(result)

}
